src/dialog/enemyeditor.cpp

```diff
@@ -0,0 +1,1479 @@
+#define MAXSTEP (local_guyref.family==eeFIRE||local_guyref.family==eeOTHER||(local_guyref.family>=eeSCRIPT01&&local_guyref.family<=eeFFRIENDLY10))?9999:256
+#define HAS_SHIELD (local_guyref.family==eeWALK||local_guyref.family==eeFIRE||local_guyref.family==eeOTHER)
+
+	char titlebuf[256];
+	sprintf(titlebuf, "Enemy %d: %s", index, guy_string[index]);
+	
+	//Family Specific
+	string turnfreqorhaltrate = "Halt Rate:";
+	if (local_guyref.family == eeKEESE || local_guyref.family == eeGHINI || local_guyref.family == eePEAHAT || local_guyref.family == eeMANHAN
+		|| local_guyref.family == eeGLEEOK || local_guyref.family == eePATRA || local_guyref.family == eeDIG) turnfreqorhaltrate = "Turn Freq:";
+
+	= Switcher(
+	w_attributes[0] = MiscAttribute(&local_guyref.misc1);
+
+
+	tf_attributes[1] = MiscAttribute(&local_guyref.misc2);
+	tf_attributes[2] = MiscAttribute(&local_guyref.misc3);
+	tf_attributes[3] = MiscAttribute(&local_guyref.misc4);
+	tf_attributes[4] = MiscAttribute(&local_guyref.misc5);
+	tf_attributes[5] = MiscAttribute(&local_guyref.misc6);
+	tf_attributes[6] = MiscAttribute(&local_guyref.misc7);
+	tf_attributes[7] = MiscAttribute(&local_guyref.misc8);
+	tf_attributes[8] = MiscAttribute(&local_guyref.misc9);
+	tf_attributes[9] = MiscAttribute(&local_guyref.misc10);
+	tf_attributes[10] = MiscAttribute(&local_guyref.misc11);
+	tf_attributes[11] = MiscAttribute(&local_guyref.misc12);
+	tf_attributes[12] = MiscAttribute(&local_guyref.misc13);
+	tf_attributes[13] = MiscAttribute(&local_guyref.misc14);
+	tf_attributes[14] = MiscAttribute(&local_guyref.misc15);
+	tf_attributes[15] = MiscAttribute(&local_guyref.misc16);
+	tf_attributes[16] = MiscAttribute(&local_guyref.misc17);
+	tf_attributes[17] = MiscAttribute(&local_guyref.misc18);
+	tf_attributes[18] = MiscAttribute(&local_guyref.misc19);
+	tf_attributes[19] = MiscAttribute(&local_guyref.misc20);
+	tf_attributes[20] = MiscAttribute(&local_guyref.misc21);
+	tf_attributes[21] = MiscAttribute(&local_guyref.misc22);
+	tf_attributes[22] = MiscAttribute(&local_guyref.misc23);
+	tf_attributes[23] = MiscAttribute(&local_guyref.misc24);
+	tf_attributes[24] = MiscAttribute(&local_guyref.misc25);
+	tf_attributes[25] = MiscAttribute(&local_guyref.misc26);
+	tf_attributes[26] = MiscAttribute(&local_guyref.misc27);
+	tf_attributes[27] = MiscAttribute(&local_guyref.misc28);
+	tf_attributes[28] = MiscAttribute(&local_guyref.misc29);
+	tf_attributes[29] = MiscAttribute(&local_guyref.misc30);
+	tf_attributes[30] = MiscAttribute(&local_guyref.misc31);
+	tf_attributes[31] = MiscAttribute(&local_guyref.misc32);
+	
+	switch (local_guyref.family)
+	{
+		case eeWALK:
+			ddl_attributes[0] = DropDownField(&local_guyref.misc1, list_walkmisc1);
+			ddl_attributes[1] = DropDownField(&local_guyref.misc2, list_walkmisc2);
+			ddl_attributes[6] = DropDownField(&local_guyref.misc7, list_walkmisc7);
+			ddl_attributes[8] = DropDownField(&local_guyref.misc9, list_walkmisc9);
```

Comment:
(needs the misc refactor)
You'd need to do this type of thing for all the dropdowns. You don't need to do anything for textfields, unless you want to change their min/max vals or something.
```suggestion
			ddl_attributes[0]->setSelectedValue(local_guyref.attributes[0]);
			ddl_attributes[0]->setListData(list_walkmisc1);
			sw_attributes[0]->switchTo(1);  // change this 1 to a constant representing the dropdown spot in the switcher
			ddl_attributes[1]->setSelectedValue(local_guyref.attributes[1]
			ddl_attributes[1]->setListData(list_walkmisc2);
			sw_attributes[1]->switchTo(1);  // change this 1 to a constant representing the dropdown spot in the switcher
			ddl_attributes[6]->setSelectedValue(local_guyref.attributes[6]);
			ddl_attributes[6]->setListData(list_walkmisc7);
			sw_attributes[6]->switchTo(1);  // change this 1 to a constant representing the dropdown spot in the switcher
			ddl_attributes[8]->setSelectedValue(local_guyref.attributes[8]);
			ddl_attributes[8]->setListData(list_walkmisc9);
			sw_attributes[8]->switchTo(1);  // change this 1 to a constant representing the dropdown spot in the switcher
```